### PR TITLE
Fix get_integer_part for the case that `expr` is 0

### PR DESCRIPTION
**Pull request by theHamsta**
#### References to other Issues or PRs
Fix issue #17681 

#### Brief description of what is fixed or changed

When trying to evaluate an expression that was implicitly 0 I got an error message that get_integer_part did return four return values instead of two.

The reason is that the early return of `get_integer_part` was ignoring the function parameter `return_ints`
```
        # ... or maybe the expression was exactly zero
            return None, None, None, None
```
A fix would be

```
        # ... or maybe the expression was exactly zero
        if return_ints:
            return 0, 0
        else:
            return None, None, None, None
```

#### Release Notes

* core
  * Fix `get_integer_part` for the case `expr=S(0)`, `return_ints=True` (should return `0,0` instead of `None, None, None, None`)

### CODE REVIEW

**Comment by jksuom** [comment]
This seems to be the only place where `return_ints` is set.
https://github.com/sympy/sympy/blob/d57f3b772c40e960cf62a6761a38848517026859/sympy/functions/elementary/integers.py#L59-L60
Can you add a test where an error will be raised in current master?

**Comment by theHamsta** [comment]
I don't know how ill-formed this example is:
```python
def test_issue_17681():
    class identity_func(Function):

        def __new__(cls, *args, **kwargs):
            return super().__new__(cls,  *args, **kwargs)

        def _eval_evalf(self, *args, **kwargs):
            return self.args[0].evalf(*args, **kwargs)

    assert floor(identity_func(S(0)))
    assert get_integer_part(S(0), 1, {}, True) == (0, 0)
```

But this triggers the error on both asserts

**Comment by jksuom** [comment]
It looks like `get_integer_part(npart, cls._dir, {}, return_ints=True)` is never called with zero `npart`, so there will be no error even without this fix.  Perhaps it is not necessary to add examples that do not appear naturally.

**Comment by theHamsta** [comment]
Well, I encountered this bug when using SymPy with a function that happened to evaluate to zero. So this bug occurs in reality.

I don't think that to much testing can harm. Even if it is really edge cases.

**Comment by jksuom** [comment]
I agree that it is a bug, though the error does not appear in current master, and it should be fixed. However, the above example `identity_function` should be modified to work also in Python 2. (`super(Function, cls).__new__(cls, *args, **kwargs)`) And the line 506 should probably contain `identity_func`.

**Comment by theHamsta** [comment]
Thanks for the hint. Apparently, I was experimenting to much. So both assert should fail on current master but pass with the fix.

File: sympy/core/tests/test_evalf.py [hunk 1]
```diff
@@ -494,6 +494,19 @@ def test_issue_8853():
     assert get_integer_part(Rational(-1, 2), 1, {}, True) == (0, 0)
 
 
+def test_issue_17681():
+    class identity_func(Function):
+
+        def __new__(cls, *args, **kwargs):
+            return super(identity_func, cls).__new__(cls,  *args, **kwargs)
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_evalf.py]: I realise that this is just the test code but there are a lot of methods like this around the codebase so I feel I need to point out that this method is not needed.

If you didn't define `__new__` then the superclass `__new__` would have been called anyway so this method has no useful effect.

**Comment by jksuom** [comment]
Thanks, this is ready to be merged.

**Pull request merged**